Homework №3 from 21.09.2022/task2.c

```diff
@@ -0,0 +1,149 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+#define MAXNUM 100
+#define lenArrForTest 9
+
+void printArr(int *array, int lenOfArray)
+{
+    for (int i = 0; i < lenOfArray; ++i)
+    {
+        printf("%d ", array[i]);
+    }
+    printf("\n");
+}
+
+void swap(int *firstElement, int *secondElement)
+{
+    int temp = *secondElement;
+    *secondElement = *firstElement;
+    *firstElement = temp;
+}
+
+int divide(int *array, int left, int right)
+{
+    int changingElem = left;
+    for (int i = left; i < right; ++i)
+    {
+        if (array[i] < array[right])
+        {
+            swap(&array[changingElem], &array[i]);
+            changingElem += 1;
+        }
+    }
+    swap(&array[right], &array[changingElem]);
+    return changingElem;
+}
+
+void quickSortRec(int *array, int left, int right)
+{
+    if (left < right)
+    {
+        int centerElem = divide(array, left, right);
+        quickSortRec(array, left, centerElem - 1);
+        quickSortRec(array, centerElem + 1, right);
+    }
+}
+
+void quickSort(int *array, int lenOfArray)
+{
+    quickSortRec(array, 0, lenOfArray - 1);
+}
+
+void genOfArray(int *array, int lenOfArray)
+{
+    for (int i = 0; i < lenOfArray; ++i)
+    {
+        array[i] = (rand() % MAXNUM) * (1 + (rand() % 2) * (-2));
+    }
+}
+
+int binSearch(int *array, int searchElem, int left, int right)
```

Comment:
```suggestion
bool binSearch(int *array, int searchElem, int left, int right)
```
Эта функция говорит "да" или "нет". Поэтому тип результата --- `bool`.